src/market/demand.ts

```diff
@@ -14,19 +14,74 @@ export interface DemandDetails {
   properties: Array<{ key: string; value: string | number | boolean }>;
   constraints: Array<string>;
 }
+
 /**
  * @hidden
  */
 export interface DemandOptions {
   subnetTag?: string;
   yagnaOptions?: YagnaOptions;
-  marketTimeout?: number;
+
+  /**
+   * @deprecated Use `expiration` instead
+   */
   marketOfferExpiration?: number;
+
+  /**
+   * Determines the expiration time of the offer and the resulting activity in milliseconds.
+   *
+   * The value of this field is used to define how long the demand is valid for yagna to match against.
+   * In addition, it will determine how long the resulting activity will be active.
+   *
+   * For example: if `expiration` is set to 10 minutes, the demand was created and starting an activity
+   * required 2 minutes, this means that the activity will be running for 8 more minutes, and then will get terminated.
+   *
+   * **IMPORTANT**
+   *
+   * It is possible that a provider will reject engaging with that demand if it's configured  without using a deadline.
+   *
+   * **GUIDE**
+   *
+   * If your activity is about to operate for 5-30 min, {@link expiration} is sufficient.
+   *
+   * If your activity is about to operate for 30min-10h, {@link debitNotesAcceptanceTimeoutSec} should be set as well.
+   *
+   * If your activity is about to operate longer than 10h, you need set both {@link debitNotesAcceptanceTimeoutSec} and {@link midAgreementPaymentTimeoutSec}.
+   *
+   *
+   */
+  expiration?: number;
```

Comment:
IMO it's a good idea to include the unit of time in the option name - `expirationMs`

Reply:
Good hint. I will rebase the PR to `beta` and make it BC-breaking way.